integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/SqlDWDatabricksVisitor.java

```diff
@@ -0,0 +1,125 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+
+/**
+ * {@link LogicalPlan} visitor that matches SqlDWRelation that comes from an Azure Databricks
+ * environment. This function extracts a {@link OpenLineage.Dataset} from the SQL DW/ Synapse table
+ * referenced by the relation. The convention used for a namespace is a URI of <code>
+ * sqlserver://&lt;server&gt;.&lt;.datasetId&gt;.&lt;tableName&gt;</code> . The name for Sql Dw
+ * tables may be a query (e.g. <code>
+ * ((select \"id\" FROM dbo.exampleInputA WHERE postalCode != '55555') q)</code>) or a table name
```

Comment:
Hmm, I actually don't know if this will work for Marquez. For JDBC relations, we currently just insert `COMPLEX` as the string name (admittedly ugly), with a TODO to parse the SQL to extract the table names being read - https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/LogicalRelationVisitor.java#L106-L120 . The GreatExpectations impl actually does the SQL parsing - https://github.com/OpenLineage/OpenLineage/blob/main/integration/common/openlineage/common/provider/great_expectations/action.py#L180-L186 .

Rather than posting the SQL query, I'd vote for either of the above solutions - a TODO for now (with or without the ugly substitute) or actually parse the SQL to extract the real sources.

Reply:
Thank you, Mike! I will defer to the "COMPLEX" name for now. I don't think I feel qualified to choose a T-SQL parsing engine for the project!